evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -178,15 +178,24 @@ where
                     // because we only need the from, to, value, and data. We can later change this
                     // into 1559, in the cli package, relatively easily once we
                     // know the target chain supports EIP-1559.
-                    self.broadcastable_transactions.push_back(TypedTransaction::Legacy(
-                        TransactionRequest {
-                            from: Some(broadcast.origin),
-                            to: Some(NameOrAddress::Address(call.contract)),
-                            value: Some(call.transfer.value),
-                            data: Some(call.input.clone().into()),
-                            ..Default::default()
-                        },
-                    ));
+                    if !is_static {
+                        data.subroutine.load_account(broadcast.origin, data.db);
+                        let account = data.subroutine.state().get_mut(&broadcast.origin).unwrap();
+
+                        self.broadcastable_transactions.push_back(TypedTransaction::Legacy(
+                            TransactionRequest {
+                                from: Some(broadcast.origin),
+                                to: Some(NameOrAddress::Address(call.contract)),
+                                value: Some(call.transfer.value),
+                                data: Some(call.input.clone().into()),
+                                nonce: Some(account.info.nonce.into()),
+                                ..Default::default()
+                            },
+                        ));
+
+                        // call_inner does not increase nonces, so we have to do it ourselves
+                        account.info.nonce += 1;
```

Comment:
I think we need to do `subroutine.inc_nonce(addr)` otherwise it's not persisted

Reply:
I don't think persistence is not necessary in this case, since `script` makes only one call

Reply:
Currently it is definitely possible to do so. If we don't want that (to avoid people footgunning) then we should be explicit about that - however, I think it makes sense that it's a no-op in `forge test` so you can re-use deployment suites from tests